lensed_hosts/generate_lensed_hosts_sne.py

```diff
@@ -0,0 +1,444 @@
+    ai1, ai2 = ole.alphas_sie(xlc1, xlc2, phl, ql, rle, le, eshr, eang, ekpa, xi1, xi2)
+
+    yi1 = xi1 - ai1
+    yi2 = xi2 - ai2
+    #----------------------------------------------------------------------------
+
+    lensed_mag_b_u, lensed_mag_b_g, lensed_mag_b_r, lensed_mag_b_i, lensed_mag_b_z, lensed_mag_b_y, lensed_image_b = lensed_sersic_2d(xi1,xi2,yi1,yi2,srcP_b,lens_P)
+
+    os.makedirs(os.path.join(outdir,'sne_lensed_bulges'), exist_ok=True)
+
+    fits_limg_b = os.path.join(outdir,'sne_lensed_bulges/') + str(lens_P['UID_lens']) + "_" + str(rle) + "_" + str(lensed_mag_b_u) + "_" + str(lensed_mag_b_g)+ "_" + str(lensed_mag_b_r)+ "_" + str(lensed_mag_b_i) + "_" + str(lensed_mag_b_z) + "_" + str(lensed_mag_b_y) + "_bulge.fits"#\
+
+    pyfits.writeto(fits_limg_b, lensed_image_b.astype("float32"), overwrite=True)
+
+    # ----------------------------------------------------------------------------
+
+    lensed_mag_d_u, lensed_mag_d_g, lensed_mag_d_r, lensed_mag_d_i, lensed_mag_d_z, lensed_mag_d_y, lensed_image_d = lensed_sersic_2d(xi1,xi2,yi1,yi2,srcP_d,lens_P)
+
+    os.makedirs(os.path.join(outdir,'sne_lensed_disks'), exist_ok=True)
+
+    fits_limg_d = os.path.join(outdir,'sne_lensed_disks/') + str(lens_P['UID_lens']) + "_" + str(rle) + "_" + str(lensed_mag_d_u) + "_" + str(lensed_mag_d_g)+ "_" + str(lensed_mag_d_r)+ "_" + str(lensed_mag_d_i)+ "_" + str(lensed_mag_d_z)+ "_" + str(lensed_mag_d_y)+ "_disk.fits" #\
+
+    pyfits.writeto(fits_limg_d, lensed_image_d.astype("float32"), overwrite=True)
+
+    return 0
+
+
+def cross_check_with_lensed_sne(twinkles_ID):
+    """ Stack the lensed hosts and lensed points to verify the calculation
+     and make some plots.
+    Parameters:
+    -----------
+    twinkles_ID: int
+        ID that identifies a particular system
+
+    Returns:
+    -----------
+    ximgs: nimg-element data array containing position of lensed host and image number
+    yimgs: nimg-element data array containing position of lensed points and image number
+
+    """
+    sne_lens_cats = pd.read_csv(os.path.join(twinkles_data_dir,
+                                             'cosmoDC2_v1.1.4_sne_cat.csv'))
+
+    idx = sne_lens_cats['twinkles_sysno'] == twinkles_ID
+
+    imgnos = sne_lens_cats['imno'][idx]
+    nimgs = len(sne_lens_cats['imno'][idx])
+
+    ximgs = np.zeros(nimgs, dtype=float)
+    yimgs = np.zeros(nimgs, dtype=float)
+
+    ximgs[:nimgs] = sne_lens_cats['x'][idx].values
+    yimgs[:nimgs] = sne_lens_cats['y'][idx].values
+
+    return ximgs, yimgs
+
+if __name__ == '__main__':
+
+    dsx = 0.01  # pixel size per side, arcseconds
+    nnn = 1000  # number of pixels per side
+    xi1, xi2 = ole.make_r_coor(nnn, dsx)
+
+    hdulist, ahb = load_in_data_sne()
+
+    message_row = 0
+    message_freq = 50
+    for i, row in ahb.iterrows():
+        if i >= message_row:
+            print ("working on system ", i , "of", max(ahb.index))
+            message_row += message_freq
```

Comment:
I once again see multiple FITS stamps appearing per system. Is that supposed to be happening?